src/vm/threads.cpp

```diff
@@ -8341,22 +8367,56 @@ LONG Thread::GetTotalThreadPoolCompletionCount()
         // enumerate all threads, summing their local counts.
         ThreadStoreLockHolder tsl;
 
-        total = s_threadPoolCompletionCountOverflow.Load();
+        total = GetOverflowCount(overflowCount);
 
         Thread *pThread = NULL;
         while ((pThread = ThreadStore::GetAllThreadList(pThread, 0, 0)) != NULL)
         {
-            total += pThread->m_threadPoolCompletionCount;
+            total += *GetThreadLocalCountRef(pThread, threadLocalCountOffset);
         }
     }
     else
     {
-        total = s_threadPoolCompletionCountOverflow.Load();
+        total = GetWorkerThreadPoolCompletionCountOverflow();
     }
 
     return total;
 }
 
+UINT64 Thread::GetTotalThreadPoolCompletionCount()
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+
+    UINT64 total;
+    if (g_fEEStarted) //make sure we actually have a thread store
+    {
+        // make sure up-to-date thread-local counts are visible to us
+        ::FlushProcessWriteBuffers();
```

Comment:
Why do we need to do this? We are getting a stale counts by design.

Reply:
It's probably not necessary. It was there before on the path where hill climbing gets the work item completion counts. For hill climbing too it is stale counts by design.